File: backend/app/schemas/job.py
"""Job schemas."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class JobCreate(BaseModel):
    title: str
    description: str
    requirements: dict | None = None
    responsibilities: list[str] | None = None
    required_skills: list[str] | None = None
    preferred_skills: list[str] | None = None
    experience_min: int | None = None
    experience_max: int | None = None
    education_level: str | None = None
    employment_type: str | None = None
    work_mode: str | None = None
    location: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    salary_currency: str = "USD"
    benefits: list[str] | None = None
    expires_at: datetime | None = None


class JobUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    requirements: dict | None = None
    responsibilities: list[str] | None = None
    required_skills: list[str] | None = None
    preferred_skills: list[str] | None = None
    experience_min: int | None = None
    experience_max: int | None = None
    education_level: str | None = None
    employment_type: str | None = None
    work_mode: str | None = None
    location: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    salary_currency: str | None = None
    benefits: list[str] | None = None
    status: str | None = None
    expires_at: datetime | None = None


class JobResponse(BaseModel):
    id: str
    company_id: str
    title: str
    description: str
    requirements: dict | None = None
    responsibilities: list[str] | None = None
    required_skills: list[str] | None = None
    preferred_skills: list[str] | None = None
    experience_min: int | None = None
    experience_max: int | None = None
    education_level: str | None = None
    employment_type: str | None = None
    work_mode: str | None = None
    location: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    salary_currency: str = "USD"
    benefits: list[str] | None = None
    status: str = "draft"
    applications_count: int = 0
    matches_count: int = 0
    expires_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class JobListResponse(BaseModel):
    id: str
    company_id: str
    title: str
    required_skills: list[str] | None = None
    employment_type: str | None = None
    work_mode: str | None = None
    location: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    status: str = "draft"
    matches_count: int = 0
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/lib/types/job.ts
export interface Job {
  id: string;
  company_id: string;
  title: string;
  description: string;
  requirements?: Record<string, unknown>;
  responsibilities?: string[];
  required_skills?: string[];
  preferred_skills?: string[];
  experience_min?: number;
  experience_max?: number;
  education_level?: string;
  employment_type?: string;
  work_mode?: string;
  location?: string;
  salary_min?: number;
  salary_max?: number;
  salary_currency: string;
  benefits?: string[];
  status: string;
  applications_count: number;
  matches_count: number;
  expires_at?: string;
  created_at: string;
  updated_at: string;
}

export interface JobMatch {
  id: string;
  job_id: string;
  candidate_id: string;
  vector_score?: number;
  keyword_score?: number;
  hybrid_score?: number;
  rerank_score?: number;
  final_score?: number;
  rank?: number;
  match_details?: Record<string, unknown>;
  strengths?: Record<string, unknown>;
  gaps?: Record<string, unknown>;
  interview_questions?: Record<string, unknown>;
  status: string;
  recruiter_notes?: string;
  created_at: string;
}
